benchmarks/benchmarks/signal.py

```diff
@@ -101,25 +101,46 @@ class Convolve(Benchmark):
     def setup(self, mode):
         np.random.seed(1234)
         # sample a bunch of pairs of 2d arrays
-        pairs = []
+        pairs = {'1d': [], '2d': []}
         for ma, nb in product((1, 2, 8, 13, 30, 36, 50, 75), repeat=2):
             a = np.random.randn(ma)
             b = np.random.randn(nb)
-            pairs.append((a, b))
+            pairs['1d'].append((a, b))
+
+        for n_image in [256, 512, 1024]:
+            for n_kernel in [3, 5, 7]:
+                x = np.random.randn(n_image, n_image)
+                h = np.random.randn(n_kernel, n_kernel)
+                pairs['2d'].append((x, h))
         self.pairs = pairs
 
     def time_convolve(self, mode):
-        for a, b in self.pairs:
+        for a, b in self.pairs['1d']:
             if b.shape[0] > a.shape[0]:
                 continue
             signal.convolve(a, b, mode=mode)
 
+    def time_convolve2d(self, mode):
+        for a, b in self.pairs['2d']:
+            if mode == 'valid':
+                if b.shape[0] > a.shape[0] or b.shape[1] > a.shape[1]:
+                    continue
+                continue
```

Comment:
The logic here doesn't look quite right. Remove the second `continue`?

Reply:
Whoops. Thanks for the catch. Fixed.